packages/auth/src/accessToken/routes.ts

```diff
@@ -101,22 +105,37 @@ async function rotateToken(
   ctx: RotateContext
 ): Promise<void> {
   const { id: managementId } = ctx.params
-  const token = (ctx.headers['authorization'] ?? '').replace('GNAP ', '')
+  const tokenValue = (ctx.headers['authorization'] ?? '').replace('GNAP ', '')
+
+  const trx = await AccessToken.startTransaction()
+
+  let accessItems: Access[]
+  let newToken: AccessToken | undefined
 
-  let newToken
   try {
-    newToken = await deps.accessTokenService.rotate(managementId, token)
-  } catch (error) {
-    const errorMessage = 'Could not rotate token'
-    deps.logger.error({ error: error && error['message'] }, errorMessage)
-    ctx.throw(400, { message: errorMessage })
-  }
+    newToken = await deps.accessTokenService.rotate({
+      managementId,
+      tokenValue,
+      trx
+    })
 
-  if (!newToken) {
-    ctx.throw(404, { message: 'Token not found' })
-  }
+    if (!newToken) {
+      ctx.throw(404, { message: 'Token not found' })
+    }
+
+    await deps.grantService.lock(newToken.grantId, trx)
```

Comment:
:thinking: does this need to happen before `rotate`?
(it'd require an initial `accessTokenService.get` or `getByManagementId` to get the `grantId`)

Reply:
I wonder if we should `accessToken` with its `grant` it in the context via `tokenHttpsigMiddleware`:

https://github.com/interledger/rafiki/blob/6a709dc92e8bc48ed5e173ac9fa0d908019cbab3/packages/auth/src/signature/middleware.ts#L110-L144

Reply:
yes :rocket:

Reply:
In general, do you think it's worth it to fetch the grantId again after first checking it in the middleware?

Reply:
In that case, since we will have the `grantId`, first thing would be to lock, then rotate (I don't mind just keeping the `accessItems = await deps.accessService.getByGrant(newToken.grantId, trx)` lookup)

Reply:
> Didn't refresh before commenting again
I did the same again 😆

Reply:
> Could rotate even be replaced with a delete and create?

I like this